resources/data_models/orders.yaml

```diff
@@ -0,0 +1,19 @@
+name: Orders
+required_dimensions:
+  - Income
+  - Timestamp
+optional_dimensions:
+  - Quantity
+  - User
+  - Product
+  - Source
+  - NumOrders
+metrics:
+  TotalIncome:
+    aggregation [sum, [dimensions, Income]]
+segments:
+  ReturningUsers:
+    filter: [>, [dimension, NumOrders],  1]
+breakout_dimensions:
```

Comment:
minor nit, but if we're going to have two sets of dimensions in one group above `required_dimensions` and `optional_dimensions`, we should group this with them.

Reply:
I'm not convinced. The `required_dimensions` and `optional_dimensions` are specifications about the data shape,  `breakout_dimensions`, `metrics`, & `segments` are what you can do with this. Could see having a better name though.

Reply:
One learning lesson from writing a bunch of these yamls for xrays & etl is that you want the simplest shape possible both for writing as such, and to make validation more comprehensible

Reply:
I'm really squeamish about the weird separation of a fingerprint to match (eg, required + optional dimensions) + a seemingly super-specific interactive description of breakout dimensions. 

I.e. if we have a usage (say micro x-rays) where we want 3 breakout dimensions, but normally have 5, do we edit this? 

If this isn't an x-ray specific matter, then it gets even weirder for me. Are we really saying the model defines the only dimensions that can be broken out into? Are these meant to be a suggestions? Is it just whether they are a primary (vs secondary dimension)? 

Also, there will be tables where one of the fields is a metric (eg a rollup table).

Reply:
It's meant to be more than just for x-rays, but also for things like drilldowns etc. If you want less, you could just `(take 3)` and not meant as only these, but rather, if you don't know what, these are good. Having said that, I can see the argument, that in most cases, you can assume that for the most part these will equal to fields with `special_type` that is some derivative of `type/category`, and the rest of the cases are pretty context specific. 

As for the metric thing, that's already covered (but not explicit enough), as if you look at the MBTL, dimension can stand for a field (via it's `special_type` or `name`), metric, or expression. The reasoning being that one question's metric is a subsequent question's field[_literal].